### ISSUE TITLE
depth of directory dependency graphs

### ISSUE THREAD

**Issue by dhebbeker** [doxygen/doxygen#7663]
I would like to have more depth in directory dependency graphs. 

Example setup
-------------------

 - `src/a/aa/dependent.h` includes `src/b/bb/dependency.h`
 - [Minimal example](https://github.com/doxygen/doxygen/files/4368980/doxygen-test3.zip)
 - looking at the directory dependency graph for `src` as seen in the page "src Directory Reference" (`html/dir_68267d1309a1af8e8297ef4c3efbcdba.html`; File List → src)
 - Doxygen 1.8.17 (b5fa3cd1c6e6240e20d3b80a70e3f04040b32021*)

Current output
-------------------
![dir_68267d1309a1af8e8297ef4c3efbcdba_dep](https://user-images.githubusercontent.com/1594340/77312011-77c4c300-6d01-11ea-9d68-abecf6d34c67.png) 

Hoped-for output
-----------------------
I would like to be able to see the directories `aa` and `bb` as well in the graph for `src` for example like this: 

![dir_a dot](https://user-images.githubusercontent.com/1594340/78419589-72dc0980-7647-11ea-98be-f4c340b1c1f2.png)

This is the DOT code I *manually* wrote:
```
digraph "src/a" {
  compound=true
  node [ fontsize="10", fontname="Helvetica"];
  edge [ labelfontsize="10", labelfontname="Helvetica"];
  subgraph clusterdir_src {
    graph [ bgcolor="#ddddee", pencolor="black", label="src" fontname="Helvetica", fontsize="10"]
    subgraph clusterdir_a {
      graph [ bgcolor="#eeeeff", pencolor="black", label=""];
      dir_a [shape=plaintext label="a"];
      dir_aa [shape=box label="aa" color="black" fillcolor="white" style="filled"];
    }
    subgraph clusterdir_b {
      graph [ bgcolor="#eeeeff", pencolor="black", label=""];
      dir_b [shape=plaintext label="b"];
      dir_bb [shape=box label="bb" color="black" fillcolor="white" style="filled"];
    }
  }
  
  dir_aa->dir_bb [headlabel="1", labeldistance=1.5 headhref="dir_000002_000003.html"];
}
```

---

How can this be achieved?

**Comment by albert-github** [comment]
It would be good to mention the page where you see the issue (like html/dir_68267d1309a1af8e8297ef4c3efbcdba.html or html/dir_85293cd6fb67eb83825626cd83b71cea.html) so it is easier to localize where you would like to have changes.

This issue is probably closely related to the problem, although different, as indicated in #7653.

**Comment by dhebbeker** [comment]
Ok, added information to issue description.

**Comment by dhebbeker** [comment]
From looking at the code I have the impression, that this behavior is intended by the current design as only [sub-directories](https://github.com/doxygen/doxygen/blob/30887ace1d06cc76bdd1b5aa3895e21dc39fddab/src/dotdirdeps.cpp#L57) and [parent directories](https://github.com/doxygen/doxygen/blob/30887ace1d06cc76bdd1b5aa3895e21dc39fddab/src/dotdirdeps.cpp#L39) (relative to the current directory) are added to the graph. This would explain the before mentioned graph (`aa` and `bb` are sub-sub-directories). 

I would like to generate graphs with more than the first layer of sub-directories.

**Comment by dhebbeker** [comment]
I added a graph of the desired output to the issue description.

**Comment by albert-github** [comment]
The original pull request #7687 has been squashed and refactored and integrated into master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
